code/game/objects/structures/tank_dispenser.dm

```diff
@@ -47,29 +69,27 @@
 
 /obj/structure/dispenser/ui_data(user)
 	var/list/data = list()
-	data["o_tanks"] = oxygentanks
-	data["p_tanks"] = plasmatanks
+	data["o_tanks"] = stored_oxygen_tanks.len
+	data["p_tanks"] = stored_plasma_tanks.len
 	return data
 
 /obj/structure/dispenser/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/tank/oxygen) || istype(I, /obj/item/tank/air) || istype(I, /obj/item/tank/anesthetic))
-		if(oxygentanks < 10)
+		if(stored_oxygen_tanks.len < 10)
 			user.drop_item()
 			I.forceMove(src)
-			oxytanks.Add(I)
-			oxygentanks++
+			stored_oxygen_tanks.Add(I)
 			update_icon()
 			to_chat(user, "<span class='notice'>You put [I] in [src].</span>")
 		else
 			to_chat(user, "<span class='notice'>[src] is full.</span>")
 		SSnanoui.update_uis(src)
 		return
 	if(istype(I, /obj/item/tank/plasma))
-		if(plasmatanks < 10)
+		if(stored_plasma_tanks.len < 10)
 			user.drop_item()
```

Comment:
OH fun stuff! Antidrop makes it so that you can't drop it and thus dupes it here most likely. Might be wise to put a fix for that in this PR as well since you're touching most of the code around it

Reply:
Ugh antidrop checks, I think that's just doing `if(user.drop_item())`, then continuing on as normal right?

Reply:
Yeah and a to_chat letting the player know why it failed if it fails